spec/ddtrace/contrib/http/circuit_breaker_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'spec_helper'
+
+require 'ddtrace'
+require 'ddtrace/contrib/http/circuit_breaker'
+
+RSpec.describe Datadog::Contrib::HTTP::CircuitBreaker do
```

Comment:
Should we also test actual undesired behavior?
I.e. when transport is called and http integration is on -  we don't produce any extra traces/spans.

Reply:
We could do this in a much fuller integration test, if we orchestrated all the components and asserted on the trace output... it would probably be a good idea to have this.

Only issue is its a bit tricky to coordinate the threads so that we have a reliable test that runs all the way through the HTTP layer, too. But I think we can do it, just not in this file. This is where the library would benefit from a better "integration test" suite that can black box test these kinds of things.

Reply:
Okay, I added what amounts to a fairly simple integration test for this... it activates real instrumentation for all HTTP libraries and asserts that sending traces does not produce traces for any `Datadog::Tracer` by asserting `expect_any_instance_of(Datadog::Tracer).to_not receive(:trace)`.

It then tests this against the 3 different kinds of transports today, and against all our HTTP integrations (not just Net/HTTP, in case the adapter changes.) I feel a lot more comfortable with this kind of test in place.